mongodbatlas/resource_mongodbatlas_project_api_key.go

```diff
@@ -97,6 +157,51 @@ func resourceMongoDBAtlasProjectAPIKeyRead(ctx context.Context, d *schema.Resour
 	projectID := ids["project_id"]
 	apiKeyID := ids["api_key_id"]
 
+	options := &matlas.ListOptions{}
+
+	apiKeyOrgList, _, err := conn.Root.List(ctx, options)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error getting api key information: %s", err))
+	}
+
+	projectAssignments := []APIProjectAssignmentKeyInput{}
+
+	for idx, role := range apiKeyOrgList.APIKey.Roles {
+		if strings.HasPrefix(role.RoleName, "ORG_") {
+			orgKeys, _, err := conn.APIKeys.List(ctx, apiKeyOrgList.APIKey.Roles[idx].OrgID, options)
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("error getting api key information: %s", err))
+			}
+			for _, val := range orgKeys {
+				if val.ID == apiKeyID {
+					for _, r := range val.Roles {
+						temp := new(APIProjectAssignmentKeyInput)
+						if strings.HasPrefix(r.RoleName, "GROUP_") {
+							temp.ProjectID = r.GroupID
+							for _, l := range val.Roles {
+								if l.GroupID == temp.ProjectID {
+									temp.RoleNames = append(temp.RoleNames, l.RoleName)
+								}
+							}
+							projectAssignments = append(projectAssignments, *temp)
+						}
+					}
+				}
+			}
+			break
+		}
+	}
```

Comment:
this piece of code seems to be duplicated in other parts of this file. Could you extract it in a function?